src/import.rs

```
@@ -17,6 +18,24 @@ pub fn import<R: Read>(r: R) -> Result<Vec<Task>> {
         })
 }
 
+/// Import a single JSON-formatted Task
+pub fn import_task(s : &str) -> Result<Task> {
+    serde_json::from_str(s)
+        .map_err(|e| {
+            TaskError::new(TaskErrorKind::ParserError, Some(Box::new(e)))
+        })
+}
+
+/// Reads line by line and tries to parse a task-object per line.
+pub fn import_tasks<BR: BufRead>(r : BR) -> Result<Vec<Task>> {
+    let vt = r.lines().filter_map(|l| l.ok())
+        .filter(|s| s.len() > 0)
+        .map(|s| import_task(s.as_str()))
+        .filter_map(|t| t.ok())
+        .collect();
```

I dont really think this would work. as i see it, you return None if an elem (a String) has a length greater 0, and try to call import_task on an empty String. Even if i reversed the if/else blocks, It still filters out any Result::Errs without returning them to the caller, so not much different from mine. I guess the right approach might be:
```rust
for line in r.lines() {
    let mut tasks = Vec::new();
    let line = match line {
        Ok(val) => val,
        Err(e) => return TaskError::new(TaskErrorKind::ReadError, Some(Box::new(e)))
    }
    tasks.push(match import_task(line.as_str) {
        Ok(val) => val,
        Err(e) => { /* transform to TaskError and return */ }
    });
    Ok(tasks)
}
```
With this we would have the all-or-nothing i think is appropriate, and Errors arent simple swallowed by Result::ok()/Result::and_then() or similar.
If you insist on a Vec<Result<Task>>, doing it with and_then seems like the right way to me.